### DISCUSSION TITLE
Identify the language of a piece of string instead of a file

### DISCUSSION THREAD

**Discussion by sh-soltanpour** [github-linguist/linguist#6256] [Q&A] [answered]
Hi, 

I have a use case in which I'm gonna detect the language of a string that I have (it has no file and extension information). I was wondering if Linguist is able to do so or not since I did not find such functionality in the README. 

Thanks

**Comment by lildude** [comment, marked as answer]
No, it can't directly as it stands as it is designed to analyse files as a whole. However, you should be able to tap into the various [strategies](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist.rb#L62-L71) that analyse the content in your own code. How you do that it entirely an exercise for you 😉.
